nav2_collision_monitor/src/pointcloud.cpp

```diff
@@ -78,30 +79,25 @@ bool PointCloud::getData(
     return false;
   }
 
-  tf2::Transform tf_transform;
-  if (base_shift_correction_) {
-    // Obtaining the transform to get data from source frame and time where it was received
-    // to the base frame and current time
-    if (
-      !nav2_util::getTransform(
-        data_->header.frame_id, data_->header.stamp,
-        base_frame_id_, curr_time, global_frame_id_,
-        transform_tolerance_, tf_buffer_, tf_transform))
-    {
-      return false;
-    }
-  } else {
-    // Obtaining the transform to get data from source frame to base frame without time shift
-    // considered. Less accurate but much more faster option not dependent on state estimation
-    // frames.
-    if (
-      !nav2_util::getTransform(
-        data_->header.frame_id, base_frame_id_,
-        transform_tolerance_, tf_buffer_, tf_transform))
-    {
-      return false;
-    }
-  }
+  const auto tf_transform = [&]() -> std::optional<tf2::Transform> {
```

Comment:
I don't see the value in this pattern

Reply:
The only real benefit is that *I* can read it better. Performance wise it's kind of the same. The problem of pushing this into the util, is that it will always spit an optional value, so the caller always needs to check whether there is a meaningful value there or not

Reply:
I think I'm correct in saying that the PC2/Range/Scan lines are identical. That attribute by itself makes me think that if we're making change, it should be abstracted into a function in the collision monitor's `Source` for them all to share use of. Especially considering `Source` already has the base/global frame, TF buffer, and timeout parameter locally so it would only need to pass in the current stamp/frame of interest. 

At that point, I might argue that `bool myMethodName(args..., tf2::Transform & output_transform)` would be stylistically more Nav2, but I'm not that picky. If you want to use `optional` instead, power to you, but that function should probably be abstracted if its just the same lines repeated 3x as a lambda.

Reply:
I also spotted this but went more conservative and tried not to make that many changes *besides renaming*. If you would like me to shrink this code, I'm happy to!

Reply:
Please :-)